mcux/mcux-sdk/CMSIS/Core_AArch64/Include/core_common.h

```diff
@@ -69,7 +69,9 @@
 #define GET_EL(_mode)		(((_mode) >> MODE_EL_SHIFT) & MODE_EL_MASK)
 
 /* MPIDR */
+#ifndef MPIDR_AFFLVL_MASK
 #define MPIDR_AFFLVL_MASK	(0xfful)
+#endif
```

Comment:
This was discussed and if I remember correctly the consensus was that we acknowledged that the world revolves around Zephyr so the guards should be in the modules, so I think this is it.

Reply:
this is not allowed, if Zephyr has it, you need to fix your code. It may be acceptable to use Zephyr's definition as a hack/workaround. Please revert this patch.

Reply:
also... AFAIK arm64 port doesn't use CMSIS at all, what's going on here? @carlocaione

Reply:
@gmarull this is excluding the module definition, so the zephyr one is used, is that not how it should work?

The bit in the doc is: https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-3-macro-name-collisions

"Given that Zephyr uses a fork of the corresponding upstream for each module, it is always possible to patch the macro implementation in each module to avoid collisions."

Reply:
the problem with `ifndef` is that the expanded define depends on the include order, in that case though, if Zephyr comes first by chance we should be fine, that's true.

Reply:
ok, then let's go with this

Reply:
Thanks @fabiobaltieri, @gmarull  and @mmahadevan108 to fix this issue, if fact, I lost one patch when upstream harpoon v2.4 release, so caused this building warning, sorry for that, now I created a PR for this fix, please help to review, thanks.
https://github.com/zephyrproject-rtos/hal_nxp/pull/283